data/org.eclipse.birt.report.data.oda.jdbc.ui/src/org/eclipse/birt/report/data/oda/jdbc/ui/editors/SQLDataSetEditorPage.java

```diff
@@ -1075,4 +1119,149 @@ private void insertTreeItemText() {
 			insertText(text);
 		}
 	}
+
+	private void setupFindQueryTextBox(Group group) {
+		this.findQueryText = new Text(group, SWT.BORDER);
+		this.findQueryText.setToolTipText(JdbcPlugin.getResourceString("tablepage.querytext.find.text.tooltip"));
+		GridData findQueryTextData = new GridData(GridData.FILL_HORIZONTAL);
+		findQueryTextData.horizontalSpan = 2;
+		this.findQueryText.setLayoutData(findQueryTextData);
+		// add support of additional accelerated key
+		findQueryText.addKeyListener(new KeyListener() {
+
+			@Override
+			public void keyPressed(KeyEvent e) {
+				if (isFindQueryTextForward(e)) {
+					findQueryTextForward();
+				} else if (isFindQueryTextBackward(e)) {
+					findQueryTextBackward();
+				}
+			}
+
+			private boolean isFindQueryTextForward(KeyEvent e) {
+				// CTRL + f
+				return ((e.stateMask & SWT.CONTROL) > 0) && ((e.keyCode == 'o') || (e.keyCode == 'O'));
+			}
+
+			private boolean isFindQueryTextBackward(KeyEvent e) {
+				// CTRL + SHIFT + f
+				return ((e.stateMask & SWT.CONTROL) > 0) && ((e.keyCode == 'b') || (e.keyCode == 'B'));
+			}
+
+			@Override
+			public void keyReleased(KeyEvent e) {
+			}
+
+		});
+
+	}
+
+	private void setupFindQueryTextOptions(Group group) {
+
+		GridData csLayoutData = new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING);
+		this.findQueryTextCaseSensitive = new Button(group, SWT.CHECK);
+		this.findQueryTextCaseSensitive
+				.setText(JdbcPlugin.getResourceString("tablepage.querytext.find.option.case.sensitive")); //$NON-NLS-1$
+		this.findQueryTextCaseSensitive
+				.setToolTipText(JdbcPlugin.getResourceString("tablepage.querytext.find.option.case.sensitive.tooltip")); //$NON-NLS-1$
+		this.findQueryTextCaseSensitive.setSelection(false);
+		this.findQueryTextCaseSensitive.setLayoutData(csLayoutData);
+		this.findQueryTextCaseSensitive.setEnabled(true);
+
+		GridData wwLayoutData = new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING);
+		this.findQueryTextWholeWord = new Button(group, SWT.CHECK);
+		this.findQueryTextWholeWord.setText(JdbcPlugin.getResourceString("tablepage.querytext.find.option.whole.word")); //$NON-NLS-1$
+		this.findQueryTextWholeWord
+				.setToolTipText(JdbcPlugin.getResourceString("tablepage.querytext.find.option.whole.word.tooltip")); //$NON-NLS-1$
+		this.findQueryTextWholeWord.setSelection(false);
+		this.findQueryTextWholeWord.setLayoutData(wwLayoutData);
+		this.findQueryTextWholeWord.setEnabled(true);
+	}
+
+	private void setupFindQueryTextButtons(Group group) {
+
+		GridData fwButtonFindTextLayoutData = new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING);
+		Button fwButtonFindText = new Button(group, SWT.BUTTON1);
+		fwButtonFindText.setText("▼ " + JdbcPlugin.getResourceString("tablepage.querytext.find.button.forward")); //$NON-NLS-1$
```

Comment:
Encoding such a literal character is asking for trouble. Better to use \u I think.

Reply:
Done, is replaced through unicode-escaped string-sequence.